### DISCUSSION TITLE
Revalidate with Infinite scroll component

### DISCUSSION THREAD

**Discussion by Natchii59** [vercel/next.js#54281] [Help]
### Summary

Hello, I would like to make Twitter (X) with Nextjs. For the homepage I use the [react-infinite-scroll-component](https://www.npmjs.com/package/react-infinite-scroll-component) npm package which allows to make an infinite scroll. I put a server action in the next function props. I have a list of tweets, with a like button on each. The problem is that when I like a tweet, it calls a server action, but in this one it does a revalidateTag, which therefore reloads the whole component. In the case where I scroll to load other tweets, when I like a tweet, it will return to the initial state, remove at the client level the tweets that I loaded by scrolling.
I don't know how to do that when I like a tweet, it only revalidates the likes of the tweet that I liked.
I also use the useOptimistic hook for the like button.

### Additional information

```js
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:52 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T8103
    Binaries:
      Node: 18.12.1
      npm: 8.19.2
      Yarn: 1.22.19
      pnpm: 8.6.7
    Relevant Packages:
      next: 13.4.19
      eslint-config-next: 13.4.19
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
    Next.js Config:
      output: N/A
```

**Comment by xawbi** [comment]
Hello, were you able to solve this problem?
